src/codeu/chat/client/commandline/Chat.java

```diff
@@ -46,10 +50,18 @@ public Chat(Context context) {
   // the system wants to exit, the function will return false.
   //
   public boolean handleCommand(String line) {
-
-    final Scanner tokens = new Scanner(line.trim());
-
-    final String command = tokens.hasNext() ? tokens.next() : "";
+    final List<String> args = new ArrayList<>();
+    final Tokenizer tokenizer = new Tokenizer(line);
+    
+    try {
+      for (String token = tokenizer.next(); token != null; token = tokenizer.next()) {
+        args.add(token);
+      }
+    } catch (IOException e){ 
+      System.out.println("ERROR! IOException caught in Chat.handleCommand().");
```

Comment:
I had something like that originally, but Yuhang wanted me to change it. Thanks for the protip!